xapian-core/include/xapian/weight.h

```diff
@@ -467,6 +467,9 @@ class XAPIAN_VISIBILITY_DEFAULT TfIdfWeight : public Weight {
        tfidf weight. */
     std::string normalizations;
 
+    /* These three enum class should use integers only upto 256 . Otherwise
+       during serialise(), the value may get truncated by static_cast() and
+       the desired result will not be achieved. */
```

Comment:
Surely 255? Sorry, I think I said 256 but that was when I was thinking we'd renumber to start at 0. At the number 256 it will overflow 1 byte.

More minor: "three enum classes" rather than "class", "up to" rather than "upto", and don't leave a space before the full stop. Also `static_cast<>` instead of `static_cast()`.

Reply:
Updated